master/internal/api.go

```diff
@@ -46,10 +46,14 @@ func (a *apiServer) paginate(p **apiv1.Pagination, values interface{}, offset, l
 		EndIndex:   endIndex,
 		Total:      total,
 	}
-	rv.Elem().Set(rv.Elem().Slice(int(startIndex), int(endIndex)))
-	return grpc.ValidateRequest(
+	err := grpc.ValidateRequest(
 		func() (bool, string) { return 0 <= startIndex && startIndex <= total, "offset out of bounds" },
```

Comment:
Is startIndex == total valid? Seems to me that would be out of bounds, and we should check startIndex < total.

Reply:
that stood out to me too but if I'm not mistaken it is. experiment ids start at 1.

{"experiments":[],"pagination":{"offset":67,"limit":0,"startIndex":67,"endIndex":67,"total":67}}

Reply:
scratch that example. endIndex is exclusive but startIndex is inclusive.